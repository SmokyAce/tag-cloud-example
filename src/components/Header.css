
.header {
    margin: 50px 50px 0;
}

.header h1 {
    font-size: 3em;
    margin: 0;
}
