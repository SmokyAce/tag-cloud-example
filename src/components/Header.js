import React from 'react';
import PropTypes from 'prop-types';
import './Header.css';

const Header = ({ children, ...props }) => {
  return (
    <div className='header' {...props}>
      <h1>{children}</h1>
    </div>
  );
};

Header.propTypes = {
  children: PropTypes.any
};

export default Header;
