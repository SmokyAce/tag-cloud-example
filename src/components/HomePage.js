import React from 'react';
import Header from './Header';
import TagCloud from './TagCloud';

const HomePage = props => {
  return (
    <React.Fragment>
      <Header>Tag Cloud</Header>
      <TagCloud {...props} />
    </React.Fragment>
  );
};

export default HomePage;
