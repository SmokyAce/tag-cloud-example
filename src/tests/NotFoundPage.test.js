/* eslint-env jest */
import React from 'react';
import ReactDOM from 'react-dom';
import TestRenderer from 'react-test-renderer';
import HomePage from '../components/NotFoundPage';
import Header from '../components/Header';

describe('<HomePage />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<HomePage />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  const testRenderer = TestRenderer.create(<HomePage />);

  it('renders correctly', () => {
    const snapshot = testRenderer.toJSON();
    expect(snapshot).toMatchSnapshot();
  });

  const testInstance = testRenderer.root;

  expect(testInstance.findByType(Header).props.children).toBe(
    '404: Page not found'
  );
});
