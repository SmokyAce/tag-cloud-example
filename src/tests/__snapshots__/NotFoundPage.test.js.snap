// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomePage /> renders correctly 1`] = `
<div
  className="header"
>
  <h1>
    404: Page not found
  </h1>
</div>
`;
